### ISSUE TITLE
Factor Weights

### ISSUE THREAD

**Issue by GWMcElfresh** [GWMcElfresh/SpectraWrapper#8]
@gboggy2 Hey Greg, I was reading the spectra paper, and something I didn't appreciate at first blush is that the factors themselves each have a weight too. I think that is what this parameter does. (Perhaps this is `SPECTRA_L`?) 

https://github.com/GWMcElfresh/SpectraWrapper/blob/9acfc3903b031ba188f63ca312a50bc9635cf9e6/scripts/run_Spectra.py#L60

I don't precisely know what emergent properties to expect (other than a weighted model definitely having a better loss function) like training speed, factor weight pliability, etc, but I'm thinking we at least expose it, and maybe later do an A/B test once this is easier to run?

**Comment by gboggy2** [comment]
It’s not SPECTRA_L, which gets output to SPECTRA_L.json and has an output like this: {"global": 103, "T_NK": 95}, indicating the numbers of factors for each of the cell type labels.

I also didn’t realize there are scores associated with the factors—we don’t capture these currently, and I don’t know the output to look for. Looking at Spectra on github, we can access the model parameters as shown here: https://github.com/dpeerlab/spectra/tree/main?tab=readme-ov-file#accessing-model-parameters

**Comment by GWMcElfresh** [comment]
Thanks - I also don't immediately see what the output would be. Reading more, it is possible they're just used internally?

Section in the methods:
>To avoid penalizing novel factors that have no relation to the annotations, we introduce a weighting matrix that scales the computation of gene–gene similarity scores by factor-specific weights that are learned from the data. Factors that have low weight are not used in computing edge probabilities, whereas factors with high weights influence the edge probabilities directly. Hence, Spectra can estimate similar parameters for two genes without forcing a high edge probability between them as long as the factors corresponding to these genes also have low weight. These weights allow the addition of new, unbiased factors that are not influenced by the input annotations. Importantly, weights are estimated from the data, allowing for an adaptive determination of the relative number of unbiased and biased factors.

My interpretation here is that there are factor specific weights, but perhaps they're only used to propose:

1. new genes that should belong in an existing factor.
2. new factors that supplement the latent structure of the data. (?)

and not exported?

**Comment by gboggy2** [comment]
That sounds like a reasonable interpretation to me. I think I captured most if not all of the useful Spectra output and these weights don’t seem to be among the output.

The output that I have the least understanding of is the overlap matrix, which I have not spent much time thinking about—but perhaps this tells us something?
[SPECTRA_overlap.csv](https://github.com/user-attachments/files/18428173/SPECTRA_overlap.csv)

**Comment by GWMcElfresh** [comment]
My initial take on the overlap matrix was that it's an overlap coefficient to see if factors were redundant, but that doesn't quite make sense if "every gene is in every factor" unless they threshold it.

back to the paper!
